### PR TITLE
Add modulemd_add_platform tool

### PR DESCRIPTION
**Pull request by ppisar**
This patch set adds a new tool "modulemd_add_platform" tool. It can be used for branching modules to a new Fedora release. I.e. it ports your modulemd-packager-v3 document from one Fedora release to another one. Technically it duplicates a context configuration section for a new platform.

Python is a new language for me, so forgive me my mistakes. I studied the latest Python, setuptools, and Fedora Packaging guidelines so the code, especially the RPM packaging uses a new syntax. I don't know how old Fedora you want to support.

I'd like see this released soon so that Fedora relengs can use it for branching modules in Fedora 37.

### CODE REVIEW

File: modulemd_add_platform/README [hunk 1]
```diff
@@ -0,0 +1,5 @@
+This tool edits a modulemd-packager YAML document.
```
Comment (FrostyX) [inline modulemd_add_platform/README:1]: Can you please add this file as `README.md` and add a heading to it, to be consistent with other tools?

**Comment by FrostyX** [comment]
Thank you @ppisar for a very nice contribution with documentation, test cases, and everything. I have only a few notes.

File: modulemd_add_platform/README (hunk 1 above)
Comment (ppisar) [reply modulemd_add_platform/README:1]: I will do it, but your advice damages Python package metadata. A content of README is included into long description. That will pollute it with markdown codes and a useless, duplicate title.

File: README.md [hunk 2]
```diff
@@ -279,6 +287,20 @@ $ grep name modules.yaml -A2
...
+Modulemd YAML files in modulemd-packager format needs to list each supported
+platform stream. When porting a module to the next distribution version it is
+necessary to add a new `context configuration` matching the new platform.
+`modulemd_add_platform` tool helps with it.
+
+For instance, porting a `module.yaml` from Fedora 35 to Fedora 36 can be
+achieved with:
+
+```
+$ modulemd_add_platform --old f35 --new f36 module.yaml
```
[the remark this answers is not in the document]
Comment (ppisar) [reply README.md:301]: I like underscores more, but I will rename everything except Python identifiers as there a hyphen is forbidden.

[the remark this answers is not in the document]
Comment (sgallagher) [reply README.md:301]: I recommend leaving them as underscores. You can double-click to select the whole name, whereas with hyphens you'll only get one section of it.

**Comment by ppisar** [comment]
I do not want to put more files than necessary. Files which are not used for anything. Have you tried copying an example from the manual page?

**Comment by ppisar** [comment]
There is already modulemd-add-platform/tests/test.yaml. ``modulemd-add-platform --old f34 --new f36 /tmp/test.yaml`` works for me on that file.

**Comment by ppisar** [comment]
I added two commits which addresses your issues.

**Comment by FrostyX** [comment]
Thank you for the PR,
I am merging and going to make a new release right away.

**Comment by ppisar** [comment]
That error message sound like you forget to copy a header (document and version lines) of the YAML document. You can verify validity of the document with "modulemd-validator --type modulemd-packager-v3 FILE".

**Pull request merged**